tools/Google.Cloud.Tools.DocUploader/Upload.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using Google.Apis.Auth.OAuth2;
+using Google.Cloud.Storage.V1;
+
+namespace Google.Cloud.Tools.DocUploader;
+
+/// <summary>
+/// Helper class to upload a file to google cloud storage.
+/// </summary>
+public class Upload
+{
+    /// <summary>
+    /// Uploads the given file to the google cloud storage.
+    /// </summary>
+    /// <param name="source">The file to be uploaded on storage bucket. Must not be null.</param>
+    /// <param name="destination">The name of the object within the bucket. Must not be null.</param>
+    /// <param name="bucketName">The name of the bucket containing the object. Must not be null.</param>
+    /// <param name="credentialsPath">The path to credentials.json file. Must not be null.</param>
+    public static void UploadFile(string source, string destination, string bucketName, string credentialsPath)
```

Comment:
Given how little code there is here, I'm not sure it's actually worth having as a separate class... particularly if we end up going with creating the tarball in memory. I'd just inline this code, or at least move it to a local method in Program.